app/code/Magento/InventoryInStorePickupShipping/Model/Carrier/Validation/SalesChannelValidator.php

```diff
@@ -74,4 +79,23 @@ public function validate(RateRequest $rateRequest): ValidationResult
 
         return $this->validationResultFactory->create(['errors' => $errors]);
     }
+
+    /**
+     * Check if at least one Pickup Location is available for the website sales channel.
+     *
+     * @param string $websiteCode
+     *
+     * @return bool
+     */
+    private function isAnyPickupLocationAvailable(string $websiteCode): bool
```

Comment:
It would be cool to extract this method into a separate service. More likely it will not be the only place where we need this logic.

Reply:
During refactoring I have found that this is the only place where this check is used.
I think we can add it in future if we need to.